test/gtest/uct/ib/test_ib.cc

```diff
@@ -445,9 +445,23 @@ class test_uct_ib_sl : public test_uct_ib_with_specific_port {
 };
 
 UCS_TEST_P(test_uct_ib_sl, check_ib_sl_config) {
+    const char *avail_sl_mask_str = getenv("GTEST_AVAIL_SL");
+    unsigned short avail_sl_mask;
+    uint8_t sl;
+
+    if (avail_sl_mask_str == NULL) {
+        avail_sl_mask = 0xffff;
+    } else {
+        avail_sl_mask = (unsigned short)strtoul(avail_sl_mask_str, NULL, 16);
```

Comment:
minor: since it would be small change, maybe add something like below as helper function:
```
int long parse_ul(const char *str, unsigned long *value) {
  int base = 10;
  if (!strncmp(str, "0x", 2)) {
    base = 16;
    str += 2;
  }
  mask = strtoul(str, &end, base);
  if (*end) {
    // err
  }
}

if (mask > UINT16_MAX
```

Reply:
Not sure we need it.
The `GTEST_AVAIL_SL` value will be passed from a script in a very specific predefined format, it has only a single user..

Reply:
up to you, but change is small, any help in spotting typos/sanitize mask could help

Reply:
IMO instead of passing a mask from the user, pass maximal SL value

Reply:
@yosefe so it means the user could not configure specific SLs but just use the first `n` SLs instead, right?

Reply:
It means we'll provide the max avail SL from the env, then inside the test we try up to this SL.

Reply:
yes, IMO it is simpler and good enough